presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -160,6 +166,13 @@ public SqlQueryExecution(QueryId queryId,
                 }
             });
 
+            stateMachine.addStateChangeListener(state -> {
```

Comment:
Moved notification to `start` method (https://github.com/prestodb/presto/pull/6114/commits/358f8dfec3d9d7be433f7a7c9930b1cf274d835c#diff-8bde286a7522eaae1ec676b3775d6122R249).

I left `endQuery` as-is because it's to be fired when query is finished so I believe we shouldn't be waiting with finishing query for this method to end, thus async is fine IMO.